subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/UnzipTransform.java

```diff
@@ -41,13 +44,15 @@
  * is located in the output directory of the transform and is named after the zipped file name
  * minus the extension.
  */
-public interface UnzipTransform extends TransformAction<TransformParameters.None> {
+@DisableCachingByDefault(because = "Not worth caching")
+public abstract class UnzipTransform implements TransformAction<TransformParameters.None> {
```

Comment:
Should we add a warning if an interface has a default implementation for `@TaskAction` or for `TransformAction.transform()`? Maybe create an issue to do this later?

Reply:
I created https://github.com/gradle/gradle/issues/17667.